packages/icon/package.py

```diff
@@ -0,0 +1,247 @@
+        XML2L_LIBS='-lxml2 '
+
+        # Blas lapack library
+        BLAS_LAPACK_LIBS='-llapack -lblas '
+
+        # Netcdf library
+        if self.compiler.name == 'cce':
+            NETCDFL='-L' + self.spec['netcdf-c'].prefix + '/crayclang/9.0/lib '
+        else:
+            NETCDFL='-L' + self.spec['netcdf-c'].prefix + '/' + self.compiler.name + '/' + str(self.compiler.version.up_to(2))  + '/lib '
+        NETCDF_LIBS='-lnetcdf -lnetcdff '
+
+        # Set MPI_LAUNCH
+        args.append('MPI_LAUNCH=false')
+
+        # CPPFLAGS
+        CPPFLAGS+=XML2I
+
+        # PGI compiler flags
+        if self.compiler.name == 'pgi':
+            # Standard cpp library
+            STDCPPL='-L/opt/gcc/8.3.0/snos/lib64 '
+            STDCPP_LIBS='-lstdc++ '
+
+            # CFLAGS
+            CFLAGS+='-g -O2 '
+
+            # FCFLAGS
+            FCFLAGS+='-g -O -Mrecursive -Mallocatable=03 '
+
+            # PGI GPU
+            if self.spec.variants['icon_target'].value == 'gpu':
+                # NVCFLAGS
+                args.append('NVCC=nvcc')
+                args.append('NVCFLAGS=--std=c++11 -arch=sm_60 -g -O3')
+                args.append('--disable-loop-exchange')
+                args.append('--enable-gpu')
+                FCFLAGS+='-acc=verystrict  -ta=nvidia:cc' + self.spec.variants['cuda_arch'].value + ' -Minfo=accel,inline '
+            # PGI CPU
+            elif self.spec.variants['icon_target'].value == 'cpu':
+                FCFLAGS+='-tp=haswell '
+
+            LDFLAGS+=STDCPPL + XML2L + NETCDFL
+            LIBS+='-Wl,--as-needed ' + XML2L_LIBS + BLAS_LAPACK_LIBS + STDCPP_LIBS + NETCDF_LIBS
+
+        # INTEL compiler flags
+        elif self.compiler.name == 'intel':
+            CFLAGS+='-g -O3 -ftz '
+            FCFLAGS+='-O2 -assume realloc_lhs -ftz '
+            LDFLAGS+=XML2L + NETCDFL
+            LIBS+='-Wl,--disable-new-dtags -Wl,--as-needed ' + XML2L_LIBS + BLAS_LAPACK_LIBS + NETCDF_LIBS
+
+        # CRAY compiler flags
+        elif self.compiler.name == 'cce':
+            CFLAGS+='-g -O3 '
+            FCFLAGS+='-hadd_paren -r am -Ktrap=divz,ovf,inv -hflex_mp=intolerant -hfp1 -O1,cache0 '
+            LDFLAGS+=XML2L
+            LIBS+='-Wl,--as-needed ' + XML2L_LIBS + BLAS_LAPACK_LIBS
+
+        args.append('CFLAGS=' + CFLAGS)
+        args.append('CPPFLAGS=' + CPPFLAGS)
+        args.append('FCFLAGS=' + FCFLAGS)
+        args.append('LDFLAGS=' + LDFLAGS)
+        args.append('LIBS=' + LIBS)
+
+        return args
+
+    def configure(self, spec, prefix):
+        if '~skip-config' in spec:
+            configure = Executable('./configure --prefix=' + prefix)
+            configure(*self.configure_args())
+
+    def install(self, spec, prefix):
+        if '~skip-config' in spec:
+            make('install')
+        else:
+            mkdir(prefix.bin)
+            install('bin/icon', prefix.bin)
+
+    @run_after('build')
+    def test(self):
```

Comment:
I think "test" is a bit misleading here. Just running an experiment is not a test, it just tells you that the binary works technically, but does not tell you anything about the output generated by it. There is a checksuite in ICON that does testing. If anything, this should be called here.

Reply:
It is misleading for the moment indeed. I might change the function to really test things later on though, that's why it's called test, but I could call it anything it is just a function called after the build phase